cmd/root.go

```diff
@@ -73,60 +73,143 @@ var (
+	// RootCmd represents the base command when called without any subcommands.
+	RootCmd := &cobra.Command{
+		Use:           "k6",
+		Short:         "a next-generation load generator",
+		Long:          BannerColor.Sprintf("\n%s", consts.Banner()),
+		SilenceUsage:  true,
+		SilenceErrors: true,
+		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+			if !cmd.Flags().Changed("log-output") {
+				if envLogOutput, ok := os.LookupEnv("K6_LOG_OUTPUT"); ok {
+					logOutput = envLogOutput
+				}
+			}
+			err := setupLoggers(ctx, logger, fallbackLogger, logFmt, logOutput)
+			if err != nil {
+				return err
 			}
-		}
-		err := setupLoggers(logger, logFmt, logOutput)
-		if err != nil {
-			return err
-		}
 
-		if noColor {
-			// TODO: figure out something else... currently, with the wrappers
-			// below, we're stripping any colors from the output after we've
-			// added them. The problem is that, besides being very inefficient,
-			// this actually also strips other special characters from the
-			// intended output, like the progressbar formatting ones, which
-			// would otherwise be fine (in a TTY).
-			//
-			// It would be much better if we avoid messing with the output and
-			// instead have a parametrized instance of the color library. It
-			// will return colored output if colors are enabled and simply
-			// return the passed input as-is (i.e. be a noop) if colors are
-			// disabled...
-			stdout.Writer = colorable.NewNonColorable(os.Stdout)
-			stderr.Writer = colorable.NewNonColorable(os.Stderr)
-		}
-		stdlog.SetOutput(logger.Writer())
-		logger.Debugf("k6 version: v%s", consts.FullVersion())
-		return nil
-	},
+			if noColor {
+				// TODO: figure out something else... currently, with the wrappers
+				// below, we're stripping any colors from the output after we've
+				// added them. The problem is that, besides being very inefficient,
+				// this actually also strips other special characters from the
+				// intended output, like the progressbar formatting ones, which
+				// would otherwise be fine (in a TTY).
+				//
+				// It would be much better if we avoid messing with the output and
+				// instead have a parametrized instance of the color library. It
+				// will return colored output if colors are enabled and simply
+				// return the passed input as-is (i.e. be a noop) if colors are
+				// disabled...
+				stdout.Writer = colorable.NewNonColorable(os.Stdout)
+				stderr.Writer = colorable.NewNonColorable(os.Stderr)
+			}
+			stdlog.SetOutput(logger.Writer())
+			logger.Debugf("k6 version: v%s", consts.FullVersion())
+			return nil
+		},
+	}
+	return RootCmd
 }
 
 // Execute adds all child commands to the root command sets flags appropriately.
 // This is called by main.main(). It only needs to happen once to the rootCmd.
-func Execute() {
+func Execute() { //nolint:funlen
+	ctx := context.Background()
```

Comment:
Why not pass this in as a parameter as well?

Reply:
Only the context? Mostly because this isn't used anywhere else but `main` and decided not to touch more files :P 

I would argue this can be left for when this actually gets some tests that will make that useful :)